### DISCUSSION TITLE
Always staying logged in (Not intended)

### DISCUSSION THREAD

**Discussion by lgtmrobo** [wevm/wagmi#462] [Q&A] [answered]
When I disconnect the wallet using **useDisconnect** the wallet remains connected on page refresh or in a new tab. Has anyone come across this issue? 

I want auth to persist only if the user remained logged in. If they disconnect and come back at a later date/time I would like it to remain logged out.

**Comment by tmm** [comment, marked as answer]
What connector(s) are you using? Programmatic disconnect does not work for most injected wallets so this behavior is simulated with the `shimDisconect` flag.

**Comment by lgtmrobo** [comment]
I'm actually using wagmi as part of RainbowKit, and using their providers. I believe this is the fix for the issue though
